src/ESA.cpp

```diff
@@ -0,0 +1,232 @@
+#include <omp.h>
+#include "ESA.hpp"
+#include <sstream>
+#include <iomanip>
+#include <cmath>
+
+using namespace std;
+
+// -----------------------------------------------------------------
+// ESA class
+// -----------------------------------------------------------------
+
+ESA::ESA(const StlsInput& input) : Stls(input) {
+   
+}
+
+// ESA compute function
+int ESA::compute(){
+  try {
+    init();  
+    if (verbose) cout << "Structural properties calculation ESA ..." << endl;
+    doESA();
+    if (verbose) cout << "Done" << endl;
+    return 0;
+  }
+  catch (const runtime_error& err) {
+    cerr << err.what() << endl;
+    return 1;
+  }
+}
+
+
+// ESA main function
+void ESA::doESA() {
+
+// Start timing
+double tic = omp_get_wtime();
+// Compute ESA static local field correction
+computeSlfc();
+// Compute static structure factor
+computeSsf();
+// End timing
+double toc = omp_get_wtime();
+// Print diagnostic
+if (verbose) {
+    printf("Elapsed time: %f seconds\n", toc - tic);
+    fflush(stdout);
+}
+  
+}
+
+
+void ESA::computeSlfc() {
+
+    // Get the degeneracy and coupling parameters
+    const double theta = in.getDegeneracy();
+    const double rs = in.getCoupling();
+
+    // Constants for the computation of the static local field correction
+    const double g0aa1 = 18.4376509088802;
```

Comment:
All the numerical constants that you define between lines 60 and 117 declare them as constexpr, it allows for some further compiler optimization because the compiler will know that this expression is never going to change. If you have problems compiling with constexpr just let me know